src/config/CompactConfig.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CompactSerializer} from '../serialization/compact/CompactSerializer';
+
+/**
+ * User-defined serialization config for the client.
+ */
+export interface CompactConfig {
+
+    /**
+     * Defines Compact serializers.
+     */
+    serializers?: Array<CompactSerializer<any>>;
+
+}
+
+/** @internal */
+export class CompactConfigImpl implements CompactConfig {
+    serializers: Array<CompactSerializer<new () => any>> = [];
```

Comment:
Changed interface to be `new () => any` instead of `any`. I think it was a leftover from the time I was researching usage with constructor functions.